src/test/scala/edu/ie3/osmogrid/lv/coordinator/LvCoordinatorSpec.scala

```diff
@@ -0,0 +1,329 @@
+ */
+
+package edu.ie3.osmogrid.lv.coordinator
+
+import akka.actor.testkit.typed.CapturedLogEvent
+import akka.actor.testkit.typed.Effect.{
+  MessageAdapter,
+  Spawned,
+  SpawnedAnonymous
+}
+import akka.actor.testkit.typed.scaladsl.{
+  ActorTestKit,
+  BehaviorTestKit,
+  ScalaTestWithActorTestKit
+}
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior}
+import edu.ie3.datamodel.models.input.connector.`type`.{
+  LineTypeInput,
+  Transformer2WTypeInput
+}
+import edu.ie3.datamodel.models.input.container.SubGridContainer
+import edu.ie3.osmogrid.cfg.OsmoGridConfigFactory
+import edu.ie3.osmogrid.exception.RequestFailedException
+import edu.ie3.osmogrid.guardian.run.RunGuardian
+import edu.ie3.osmogrid.io.input.InputDataProvider
+import edu.ie3.osmogrid.io.input.InputDataProvider.AssetInformation
+import edu.ie3.osmogrid.io.output.ResultListener.ResultEvent
+import edu.ie3.osmogrid.lv.LvRegionCoordinator.Partition
+import edu.ie3.osmogrid.lv.coordinator.MessageAdapters.{
+  WrappedInputDataResponse,
+  WrappedRegionResponse
+}
+import edu.ie3.osmogrid.lv.{LvRegionCoordinator, coordinator}
+import edu.ie3.osmogrid.model.{DummyOsmoGridModel, OsmoGridModel}
+import edu.ie3.test.common.UnitSpec
+import org.scalatest.BeforeAndAfterAll
+import org.slf4j.event.Level
+
+import java.util.UUID
+
+class LvCoordinatorSpec
+    extends ScalaTestWithActorTestKit
+    with UnitSpec
+    with BeforeAndAfterAll {
+  private val asynchronousTestKit = ActorTestKit()
+
+  "Having a lv coordinator" when {
+    val cfg = OsmoGridConfigFactory.defaultTestConfig.generation.lv.getOrElse(
+      fail("Test config does not contain config for lv grid generation.")
+    )
+    val inputDataProvider =
+      asynchronousTestKit.createTestProbe[InputDataProvider.Request](
+        "InputDataProvider"
+      )
+    val lvCoordinatorAdapter =
+      asynchronousTestKit.createTestProbe[coordinator.Response](
+        "RunGuardian"
+      )
+    val inputDataProviderAdapter =
+      asynchronousTestKit.createTestProbe[InputDataProvider.Response](
+        "InputDataProviderAdapter"
+      )
+    val regionCoordinatorAdapter =
+      asynchronousTestKit.createTestProbe[LvRegionCoordinator.Response](
+        "InputDataProviderAdapter"
+      )
+    val msgAdapters = MessageAdapters(
+      inputDataProviderAdapter.ref,
+      regionCoordinatorAdapter.ref
+    )
+
+    "being initialized" should {
+
+      "register the correct amount of message adapters" in {
+        val idleTestKit = BehaviorTestKit(
+          coordinator.LvCoordinator(
+            cfg,
+            inputDataProvider.ref,
+            lvCoordinatorAdapter.ref
+          )
+        )
+
+        idleTestKit.expectEffectType[
+          MessageAdapter[InputDataProvider.Response, coordinator.Request]
+        ]
+        idleTestKit.expectEffectType[
+          MessageAdapter[LvRegionCoordinator.Response, coordinator.Request]
+        ]
```

Comment:
Could you briefly enlighten me on how that works? 👀

Reply:
I don't know either the details "under the hood". However, when something happens within an Actor, that can be sensed as an effect. Here, the test kit re-assures, that the expected message adapters are registered with the Actor.